File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_backend/entity/procurement/SupplierEntity.java
package com.example.fieldsync_inventory_backend.entity.procurement;

import jakarta.persistence.*;
import lombok.*;
import com.example.fieldsync_inventory_backend.entity.Auditable;

import java.time.Instant;

@Entity
@Table(name = "supplier")
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SupplierEntity extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "supplier_type_id")
    private SupplierTypeEntity supplierType;

    @Column(name = "full_name", length = 100, nullable = false)
    private String fullName;

    @Column(length = 50)
    private String email;

    @Column(length = 20, nullable = false)
    private String phone;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/local/entity/SupplierEntity.kt
package com.example.fieldsync_inventory_app.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "supplier")
data class SupplierEntity(
    @PrimaryKey val id: Int,
    @ColumnInfo(name = "supplier_type_id")
    val supplierTypeId: Int,
    @ColumnInfo(name = "supplier_type_name")
    val supplierTypeName: String,
    @ColumnInfo(name = "full_name")
    val fullName: String,
    val email: String?,
    val phone: String?,
    val address: String?,
    val description: String?,
)
